homeassistant/components/hydrawise/__init__.py

```diff
@@ -64,3 +75,21 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass.data[DOMAIN].pop(entry.entry_id)
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    if config_entry.version == 1:
```

Comment:
Changing major version causes the config entry to not be backwards compatible. In 2024.1 we introduced minor version. I asked around if it's reasonable to start the reauth in the migration step or not. Otherwise I don't think a migration is needed here.

Reply:
So, if I understand the minor version correctly, will we keep the old API key in the config entry in case someone downgrades? And then increment the minor version and add the new password and username in the migration step?

Reply:
I don't think you should set the username and password empty in the entry data. You can just check on init if it's in the data (left that in another comment)

Reply:
Fixed.